Terminal.Gui/Views/FileDialog.cs

```diff
@@ -1,1053 +1,1634 @@
+				Driver.AddStr (new string (' ', feedbackPadLeft));
+				Driver.AddStr (feedback);
+				Driver.AddStr (new string (' ', feedbackPadRight));
+			}
+		}
 
-			nameEntry.TextChanged += (s,e) => {
-				if (nameEntry.Text.IsEmpty) {
-					this.prompt.Enabled = false;
-				} else {
-					this.prompt.Enabled = true;
-				}
-			};
+		/// <inheritdoc/>
+		public override void OnLoaded ()
+		{
+			base.OnLoaded ();
+			if (loaded) {
+				return;
+			}
+			loaded = true;
+
+			// May have been updated after instance was constructed
+			this.btnOk.Text = Style.OkButtonText;
+			this.btnUp.Text = this.GetUpButtonText();
+			this.btnBack.Text = this.GetBackButtonText();
+			this.btnForward.Text = this.GetForwardButtonText();
+			this.btnToggleSplitterCollapse.Text = this.GetToggleSplitterText(false);
+			
+			tbPath.Autocomplete.ColorScheme.Normal = Attribute.Make (Color.Black, tbPath.ColorScheme.Normal.Background);
+
+			treeView.AddObjects (Style.TreeRootGetter ());
+
+			// if filtering on file type is configured then create the ComboBox and establish
+			// initial filtering by extension(s)
+			if (this.AllowedTypes.Any ()) {
+
+				this.CurrentFilter = this.AllowedTypes [0];
+
+				// Fiddle factor
+				var width = this.AllowedTypes.Max (a => a.ToString ().Length) + 6;
+
+				allowedTypeMenu = new MenuBarItem ("<placeholder>",
+					allowedTypeMenuItems = AllowedTypes.Select (
+						(a, i) => new MenuItem (a.ToString (), null, () => {
+							AllowedTypeMenuClicked (i);
+						}))
+					.ToArray ());
+
+				allowedTypeMenuBar = new MenuBar (new [] { allowedTypeMenu }) {
+					Width = width,
+					Y = 1,
+					X = Pos.AnchorEnd (width),
+
+					// TODO: Does not work, if this worked then we could tab to it instead
+					// of having to hit F9
+					CanFocus = true,
+					TabStop = true
+				};
+				AllowedTypeMenuClicked (0);
+
+				allowedTypeMenuBar.Enter += (s, e) => {
+					allowedTypeMenuBar.OpenMenu (0);
+				};
+
+				allowedTypeMenuBar.DrawContentComplete += (s, e) => {
+					
+					allowedTypeMenuBar.Move (e.Rect.Width - 1, 0);
+					Driver.AddRune (Driver.DownArrow);
+
+				};
+
+				this.Add (allowedTypeMenuBar);
+			}
 
-			Width = Dim.Percent (80);
-			Height = Dim.Percent (80);
+			// if no path has been provided
+			if (this.tbPath.Text.Length <= 0) {
+				this.tbPath.Text = Environment.CurrentDirectory;
+			}
 
-			// On success, we will set this to false.
-			canceled = true;
+			// to streamline user experience and allow direct typing of paths
+			// with zero navigation we start with focus in the text box and any
+			// default/current path fully selected and ready to be overwritten
+			this.tbPath.FocusFirst ();
+			this.tbPath.SelectAll ();
 
-			KeyPress += (s, e) => {
-				if (e.KeyEvent.Key == Key.Esc) {
-					Cancel ();
-					e.Handled = true;
+			if (ustring.IsNullOrEmpty (Title)) {
+				switch (OpenMode) {
+				case OpenMode.File:
+					this.Title = $"OPEN {(MustExist ? "EXISTING " : "")}FILE";
```

Comment:
Eventually these need to be localizable.

Reply:
Fixed in [428c2ac](https://github.com/gui-cs/Terminal.Gui/pull/2259/commits/428c2acb174f9697ae0c3889ab389bce1f9adef0)